GlobalRegistrar/contract.sol

```diff
@@ -1,25 +1,65 @@
 //sol
 
-import "owned";
-
 contract NameRegister {
-	function addr(bytes32 _name) constant returns (address o_owner) {}
-	function name(address _owner) constant returns (bytes32 o_name) {}
+	function addr(bytes32 _name) constant returns (address o_owner);
+	function name(address _owner) constant returns (bytes32 o_name);
 }
 
 contract Registrar is NameRegister {
 	event Changed(bytes32 indexed name);
 	event PrimaryChanged(bytes32 indexed name, address indexed addr);
 
-	function owner(bytes32 _name) constant returns (address o_owner) {}
-	function addr(bytes32 _name) constant returns (address o_address) {}
-	function subRegistrar(bytes32 _name) constant returns (address o_subRegistrar) {}
-	function content(bytes32 _name) constant returns (bytes32 o_content) {}
+	function owner(bytes32 _name) constant returns (address o_owner);
+	function addr(bytes32 _name) constant returns (address o_address);
+	function subRegistrar(bytes32 _name) constant returns (address o_subRegistrar);
+	function content(bytes32 _name) constant returns (bytes32 o_content);
 	
-	function name(address _owner) constant returns (bytes32 o_name) {}
+	function name(address _owner) constant returns (bytes32 o_name);
+}
+
+contract AuctionSystem {
```

Comment:
Another question about separating the contracts. In the contract AuctionSystem is no function to actually bid on a name. Doesn't it belong here?
Also in the NameRegister contract I can see no data members, but isn't that the place for them? I think if you can not seperate it like that, you should put it all in one contract. For me separation makes only sense if you can reuse the basic contracts and/or can use them as they are, with other derived contracts only adding features to working stand-alone contracts.